src/main/java/roomescape/dao/jdbc/JdbcReservationDao.java

```diff
@@ -0,0 +1,69 @@
+package roomescape.dao.jdbc;
+
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.dao.ReservationDao;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTime;
+
+@Repository
+public class JdbcReservationDao implements ReservationDao {
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    public List<Reservation> findAllReservations() {
+        String sql = """
+            SELECT
+                r.id as reservation_id,
+                r.name,
+                r.date,
+                t.id as time_id,
+                t.start_at as time_value
+            FROM reservation as r
+            INNER JOIN reservation_time as t
+            ON r.time_id = t.id
+            """;
+
+        return jdbcTemplate.query(sql, createReservationMapper());
+    }
+
+    public Reservation addReservation(Reservation reservation) {
+        String sql = "INSERT INTO reservation (name, date, time_id) VALUES (?, ?, ?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(con -> {
+            PreparedStatement stmt = con.prepareStatement(sql, new String[]{"id"});
+            stmt.setString(1, reservation.getName());
+            stmt.setDate(2, Date.valueOf(reservation.getDate()));
+            stmt.setLong(3, reservation.getTime().getId());
+            return stmt;
+        }, keyHolder);
+
+        return new Reservation(keyHolder.getKey().longValue(), reservation.getName(),
+            reservation.getDate(), reservation.getTime());
+    }
+
+    public void removeReservationById(Long id) {
+        String sql = "DELETE FROM reservation WHERE id = ?";
+        jdbcTemplate.update(sql, id);
+    }
+
+    private RowMapper<Reservation> createReservationMapper() {
+        return (rs, rowNum) -> new Reservation(
+            rs.getLong("id"),
```

Comment:
```suggestion
            rs.getLong("reservation_id"),
```
SQL 쿼리 상에서는 r.id as reservation_id로 alias를 줬어요.
Mapper에서 가져올 때도 정확히 alias 이름으로 가져와야 할 것 같아요.

Reply:
Mapper에서도 time 테이블의 id를 가져오는지 reservation 테이블의 id를 가져오는지 명시적으로 변경하니 코드가 더 명확해질 수 있는 것 같습니다! 
추후 테이블 구조가 복잡해지더라도 alias 이름을 통해 충돌을 막고 오류를 막을 수 일을 것 같습니다. 👍🏻